### PR TITLE
Refactor person model to use email and name from patient

### PR DESCRIPTION
**Pull request by craigspaeth**
**Don't merge yet, waiting on Cliniko to approve us as a connected app**

This removes `name` and `email` from the person table to help with edge case issues that occur with the dual sources of truth such as when the person.email field is out of sync with Cliniko (e.g. Colton noticed a bug where it saved uppercase in the DB and lowercase in Cliniko).

To do make this simple we're going to use Cliniko's connected apps feature to let people search for a client in Cliniko instead of supporting our in-house autocomplete UI. I also updated the docs a bit to reflect the decisions around in-memory vs. gql modeling.

image

### CODE REVIEW

File: apps/api/jobs/charge-sessions/network.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const fetchClinikoData = async ({ appt }) => {
 export const fetchCoupons = async ({ patient }) => {
   const person = await Person.findByPatientId(patient.id)
   if (person && person.coupons.length >= 100) {
-    throw new Error(`Too many coupons on ${person.email}`)
+    throw new Error(`Too many coupons on ${patient.email}`)
```
Comment (craigspaeth) [inline apps/api/jobs/charge-sessions/network.js]: @TMcMeans Side doing work! I touched this file, so I'd love a 2nd code review here :)

Comment (TMcMeans) [reply apps/api/jobs/charge-sessions/network.js]: Okay, taking a second look!

Comment (TMcMeans) [reply apps/api/jobs/charge-sessions/network.js]: Looks like your pulling the email from patient and thats what we want! Woo!

File: apps/api/models/person/domain.js [hunk 2]
```diff
@@ -44,20 +44,19 @@ export const authorizeUpdate = (args, ctx) => {
...
-// convert record properties from snake_case to camelCase for a single record or list of records
-export const toModel = input => (input ? fromPG(input) : null)
+// Convert record properties from snake_case to camelCase for a single record
+// or list of records
+export const toModel = person => (person ? fromPG(person) : null)
 
-// explicity snake_case/transfer properties in case the record contains extra properties
-// that do not conform to the Person schema
+// Explicity snake_case/transfer properties in case the record contains
+// extra properties that do not conform to the Person schema
```
Comment (TMcMeans) [inline apps/api/models/person/domain.js]: I appreciate the comments sprinkled in this PR & readme to explain the data models

File: apps/api/models/person/gql.js [hunk 3]
```diff
@@ -143,7 +131,7 @@ export const createPersonUniqueByEmail = async (_root, args, ctx, ast) => {
   }
   if (sfWaitlistNote) await Patient.addSFWaitlistNote(patient, sfWaitlistNote)
   if (isReferral) await NotificationService.sendReferralNotification(person)
-  return toGql(person)
+  return toGql(person, patient)
```
Comment (lionsdwarf) [inline apps/api/models/person/gql.js]: 👍

File: apps/manage/lib/view-model/assign.js [hunk 4]
```diff
@@ -22,56 +22,6 @@ export const initialState = () => ({
  * Queries and mutations
  * -----------------------------------------------------------------------------
  */
-const searchPeople = async val => {
```
Comment (lionsdwarf) [inline apps/manage/lib/view-model/assign.js]: Nit pick: Should this file be renamed to `.../view-model/cliniko.js` ?

**Comment by craigspaeth** [comment]
@lionsdwarf you nailed it with your guess on how a connected app works. I'll take note of your nit-pick about naming and refactor next time I touch it. Sensing no blockers here, so I'll go ahead and merge to go out with the next deploy. Thanks!

**Comment by lionsdwarf** [comment]
👍

**Comment by lionsdwarf** [comment]
@craigspaeth Max pinged me about notifying them before we launch clinical-facing stuff, so let's just keep that in mind before we deploy.

**Pull request merged**